File: app/src/main/java/com/example/myappcontacts/di/contacts/ContactsComponent.java
package com.example.myappcontacts.di.contacts;


import com.example.myappcontacts.presentation.contacts.presenter.ContactsPresenter;

import dagger.Subcomponent;

@Subcomponent(modules = {ContactsModule.class})
@ContactsScope
public interface ContactsComponent {
    void inject(ContactsPresenter contactsPresenter);
}

File: app/src/main/java/com/example/myappcontacts/di/contacts/ContactsComponent.kt
package com.example.myappcontacts.di.contacts

import com.example.myappcontacts.presentation.contacts.presenter.ContactsPresenter
import dagger.Subcomponent

@Subcomponent(modules = [ContactsModule::class])
@ContactsScope
interface ContactsComponent{
    fun inject(contactsPresenter: ContactsPresenter)
}